mldsa/src/sign.c

```diff
@@ -909,12 +906,24 @@ int crypto_sign_verify_internal(const uint8_t *sig, size_t siglen,
   /* mld_unpack_sig and mld_polyvecl_chknorm signal failure through a
    * single non-zero error code that's not yet aligned with MLD_ERR_XXX.
    * Map it to MLD_ERR_FAIL explicitly. */
+
+  /* Constant-time: The signature is commonly considered public. However, we
+   * mark it as secret to make explicit which parts of the code take time
+   * depending on the signature. mld_unpack_sig uses conditional branches to
+   * unpack the hint vector h. */
+  MLD_CT_TESTING_DECLASSIFY(
+      sig + MLDSA_CTILDEBYTES + MLDSA_L * MLDSA_POLYZ_PACKEDBYTES,
+      MLDSA_POLYVECH_PACKEDBYTES);
```

Comment:
I'm not sure a public API should change the classification over user-provided data -- this can break or unexpectedly alter CT tests in integrations. Can we instead document that `verify` is _not_ CT for the (relevant parts of) the signature, and leave all classification to the tests?

Reply:
But, yes, we can move the declassification of the h-component to the tests insetead.
I do want to keep testing with the remainder of the signature marked as secret though.